cpp/src/arrow/compute/exec/exec_plan.h

```diff
@@ -328,27 +316,14 @@ class ARROW_EXPORT ExecNode {
   /// This may be called any number of times after StartProducing() succeeds.
   virtual void ResumeProducing(ExecNode* output, int32_t counter) = 0;
 
-  /// \brief Stop producing definitively to a single output
-  ///
-  /// This call is a hint that an output node has completed and is not willing
-  /// to receive any further data.
-  virtual void StopProducing(ExecNode* output) = 0;
```

Comment:
I've since learned that this is still needed.  This covers the case where a LIMIT X node is placed on one branch of a query.  It is intended to stop part of the plan but not abort the entire plan.  Do you think we can leave it in?